include/nebula/sclient/StorageClient.h

```diff
@@ -0,0 +1,105 @@
+/* Copyright (c) 2020 vesoft inc. All rights reserved.
+ *
+ * This source code is licensed under Apache 2.0 License,
+ * attached with Common Clause Condition 1.0, found in the LICENSES directory.
+ */
+
+#pragma once
+
+#include <functional>
+#include <memory>
+#include <string>
+#include <thread>
+#include <unordered_map>
+#include <vector>
+
+#include "common/datatypes/HostAddr.h"
+#include "nebula/sclient/ScanEdgeIter.h"
+#include "nebula/sclient/ScanVertexIter.h"
+
+#define DEFAULT_LIMIT 1000
+#define DEFAULT_START_TIME 0
+#define DEFAULT_END_TIME std::numeric_limits<int64_t>::max()
+
+namespace folly {
+class IOThreadPoolExecutor;
+}   // namespace folly
+
+namespace nebula {
+
+// Wrap the thrift client.
+namespace meta {
+class MetaClient;
+}   // namespace meta
+
+namespace storage {
+class GraphStorageClient;
+}   // namespace storage
+
+using MetaHostAddr = HostAddr;
+
+struct MetaHostAddrHash {
+    std::size_t operator()(const MetaHostAddr &h) const noexcept {
+        return std::hash<std::string>()(h.host + std::to_string(h.port));
+    }
+};
+
+enum class VidType : int8_t {
+    INT64,
+    FIXED_STRING,
+    UNKNOWN
+};
+
+class StorageClient {
+public:
+    explicit StorageClient(const std::vector<nebula::MetaHostAddr> &metaServers);
+    // disable copy
+    StorageClient(const StorageClient &) = delete;
+    StorageClient &operator=(const StorageClient &) = delete;
+
+    StorageClient(StorageClient &&c) noexcept;
+
+    ~StorageClient();
+
+    StorageClient &operator=(StorageClient &&c) noexcept;
+
+    VidType getSpaceVidType(std::string spaceName);
+
+    int32_t getSpaceVidLen(std::string spaceName);
+
+    std::vector<int32_t> getParts(std::string spaceName);
+
+    std::unordered_map<int32_t, std::vector<nebula::MetaHostAddr>> getPartsAlloc(
+        std::string spaceName);
+
+    std::unordered_map<int32_t, nebula::MetaHostAddr> getPartsLeader(std::string spaceName);
+
+    ScanEdgeIter scanEdgeWithPart(std::string spaceName,
```

Comment:
why only support the scan with the part?

Reply:
I just this interface currently